File: cracking_the_coding_interview/array_strings/is_unique.php
<?php

$string = 'amul';
$stringHasCharacters = [];
$unique = true;

if(strlen($string) > 26){
  $unique = false;
}
else{
  for($i = 0; $i < strlen($string) ; $i++){
      if(!isset($stringHasCharacters[$string[$i]])){
        $stringHasCharacters[$string[$i]] = 0;
      }
      else {
        $unique = false;
        break;
      }
  }
}

if($unique){
  echo $string." has unique characters.";
}
else{
  echo $string." has not unique characters.";
}

File: cracking_the_coding_interview/array_strings/golang/is_unique.go
/*
Is Unique: Implement an algorithm to determine if a string has all unique characters. What if you
cannot use additional data structures?
*/
package main

import "fmt"

func isUnique(text string) bool {

	if len(text) > 26 {
		return false
	}

	characterCount := make(map[string]int)

	for _, c := range text {
		char := string(c)
		if _, found := characterCount[char]; !found {
			characterCount[char] = 0
		}
		characterCount[char]++

		if characterCount[char] > 1 {
			return false
		}
	}

	return true

}

func main() {

	fmt.Println(isUnique("aA"))

}
